src/include/sof/ipc/common.h

```diff
@@ -29,6 +29,9 @@ struct ipc_msg;
 #define IPC_IS_SIZE_INVALID(object)					\
 	(object).hdr.size == sizeof(object) ? 0 : 1
 
+#define IPC_TAIL_IS_SIZE_INVALID(object)					\
+	(object).hdr.size + (object).ext_data_length == sizeof(object) ? 0 : 1
```

Comment:
at the very least maybe `IPC_TAIL_SIZE_IS_INVALID()`  and ideally I'd make this
```
#define IPC_TAIL_SIZE_IS_VALID(object) \
	((object).hdr.size + (object).ext_data_length == sizeof(object))
```
because in general affirmative names might be easier to work with, and the `==` comparison already returns a perfect boolean result, but the definition above already does the other way, so...

Reply:
I would rather `IPC_TAIL_IS_SIZE_INVALID` to keep the naming semantics the same as above or update the one above to match your suggestion. But i rather not deviate on this alone.

Seconded on the brackets, i thought that was weird, not sure why they were ommitted

I am not sure either, it is the same as using `!=` I am really not sure what is going on with that macro, i can update both to keep them the same

Looks like @lgirdwood was the original author of the macro. Maybe he can share some of his logic